File: C/Python/102.py
M = int(input())
A = [int(input()) for _ in range(M)]
N = int(input())
B = [int(input()) for _ in range(N)]

date = ["x"*32]
flag = True
for i in range(1,len(date)):
    if i in A and i in B:
        if flag:
            date[i] = "A"
            flag = False
        else:
            date[i] = "B"
            flag = True
    elif i in A:
        date[i] = "A"
    elif i in B:
        date[i] = "B"

for i in range(1,32):
    print(date[i])

File: C/JavaScript/102.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split("\n").map(Number);

let = idx = 0;
const M = input[idx++];
const A = input.slice(idx, idx + M);

idx += M;

const N = input[idx++];
const B = input.slice(idx, idx + N);

let date = Array(32).fill("x");
let flag = true;

for (let i = 1; i < 32; i++){
    let inA = A.includes(i);
    let inB = B.includes(i);

    if (inA && inB){
        if (flag){
            date[i] = "A";
            flag = false
        }else{
            date[i] = "B";
            flag = true;
        }
    }else if(inA){
        date[i] = "A";
    }else if(inB){
        date[i] = "B";
    }
}

for (let i = 1; i < 32; i++){
    console.log(date[i]);
}